freqtrade/persistence/models.py

```diff
@@ -504,6 +514,64 @@ def update_trade(self, order: Order) -> None:
             raise ValueError(f'Unknown order type: {order.order_type}')
         Trade.commit()
 
+    def process_sell_sub_trade(self, order: Order, is_closed: bool = True,
+                               is_non_bt: bool = True) -> None:
+        orders = (self.select_filled_orders('buy'))
+        sell_amount = order.safe_filled
+        sell_rate = order.safe_price
+        sell_stake_amount = sell_rate * sell_amount * (1 - self.fee_close)
+        if is_closed:
+            if sell_amount == self.amount:
+                self.close(sell_rate)
+                if is_non_bt:
+                    Trade.commit()
+                return
+        realized_profit = 0.0
+        profit = 0.0
+        idx = -1
+        while sell_amount:
+            b_order = orders[idx]
+            buy_amount = b_order.safe_amount_after_fee
+            avg_rate = b_order.safe_price
+            buy_rate = b_order.initial_average or b_order.price
+            if sell_amount < buy_amount:
+                amount = sell_amount
+            else:
+                idx -= 1
+                amount = buy_amount
+            if is_closed:
+                b_order.filled -= amount
+                b_order.order_update_date = datetime.now(timezone.utc)
+            sell_amount -= amount
+            profit += self.calc_profit2(avg_rate, sell_rate, amount)
+            realized_profit += self.calc_profit2(buy_rate, sell_rate, amount)
+        if is_closed:
+            b_order2 = orders[idx]
+            amount2 = b_order2.safe_amount_after_fee
+            b_order2.average = (b_order2.average * amount2 - profit / (1 + self.fee_open)) / amount2
+            b_order2.order_update_date = datetime.now(timezone.utc)
```

Comment:
This HAS to correspond to the date when the order was updated with a ccxt order response.
This is true for almost all values of the order object (that are assigned as part of the parse_ccxt method).

Anything else will be considered a bug.

Reply:
Updated